### PR TITLE
Fix target_network accessing data parallel model

### PR DESCRIPTION
**Pull request by amjltc295**
## Why do we need this PR?
* The optimizer cannot be loaded properly when using multiple GPUs

## How is it implemented?
*

### CODE REVIEW

File: src/pipeline/base_pipeline.py [hunk 1]
```diff
@@ -153,10 +157,11 @@ def _setup_optimizers(self):
         """
         self.optimizers = {}
         for name, entry in global_config['optimizers'].items():
+            model = self._get_non_parallel_model(self)
```
Comment (Nash2325138) [inline src/pipeline/base_pipeline.py]: The latter `self` is a typo?

File: src/pipeline/base_pipeline.py [hunk 2]
```diff
@@ -176,7 +181,7 @@ def _load_pretrained(self, pretrained_path):
         """ Load pretrained model not strictly """
         logger.info(f"Loading pretrained checkpoint: {pretrained_path} ...")
         checkpoint = torch.load(pretrained_path)
-        model = self.model.module if len(self.device_ids) > 1 else self.model
+        model = self._get_non_parallel_model(self)
```
Comment (Nash2325138) [inline src/pipeline/base_pipeline.py]: Unnecessary `self`?

File: src/worker/validator.py [hunk 3]
```diff
@@ -29,7 +30,9 @@ def _run_and_optimize_model(self, data):
 
         elif self.optimize_strategy == 'GAN':
             total_loss = 0
-            for network_name in self.model.network_names:
+            for optimizer_name in self.optimizers.keys():
+                self.optimizers[optimizer_name].zero_grad()
```
Comment (Nash2325138) [inline src/worker/validator.py]: These lines are fixed in another branch (will be a PR soon). Also, `Validator` does not hold the attribute `optimizers`.
(I'll remove it on rebasing)

File: src/pipeline/base_pipeline.py (hunk 2 above)
Comment (Nash2325138) [reply src/pipeline/base_pipeline.py]: Still not fixed

File: src/pipeline/base_pipeline.py (hunk 1 above)
Comment (amjltc295) [reply src/pipeline/base_pipeline.py]: Fixed

File: src/pipeline/base_pipeline.py (hunk 2 above)
Comment (amjltc295) [reply src/pipeline/base_pipeline.py]: Fixed

**Pull request merged**